src/citrine/_rest/collection.py

```diff
@@ -98,6 +104,70 @@ def list(self,
                 # properly, so we are filtering client-side.
                 pass
 
+    def list(self,
+             page: Optional[int] = None,
+             per_page: Optional[int] = None) -> Iterable[ResourceType]:
+        """
+        List all visible elements in the collection.
+
+        Leaving page and per_page as default values will yield all elements in the
+        collection, paginating over all available pages.
+
+        Parameters
+        ---------
+        page: int, optional
+            The "page" of results to list. Default is to read all pages and yield
+            all results.
+        per_page: int, optional
+            Max number of results to return per page. Default is 20.  If the page
+            parameter is specified this will limit the number of results in that
+            response.
+
+        Returns
+        -------
+        Iterable[ResourceType]
+            Resources in this collection.
+
+        """
+        # Do an initial fetch of the first page using supplied args.
+        # If more results expected, proceed with paginated calls.
+        first_page = self._fetch_page(page=page, per_page=per_page)
+        first_page_count = 0
+        first_uid = None
+        for idx, element in enumerate(first_page):
+            yield element
+
+            uid = getattr(element, 'uid', None)
+            if idx == 0 and not first_uid and uid:
+                first_uid = uid
+
+            first_page_count += 1
+
+        if page is not None or first_page_count != (per_page or DEFAULT_PER_PAGE):
```

Comment:
Cool way to make this change non-breaking 👍

Reply:
Agreed.

Unrelated, I suggest that you have `list` pass `DEFAULT_PER_PAGE` if `per_page` is not specified. Otherwise, there is the assumption that two unrelated pieces of logic align (the product API default per_page must equal the default per_page here, but they could be changed independently of each other)

Reply:
Good point.  I was leaving this up to the API but it makes sense to honor the per_page parameter.